config/environments/test.rb

```diff
@@ -47,3 +47,8 @@
 
   config.active_support.test_order = :random
 end
+
+ENV['ZENDESK_URL'] = 'https://test.support.zendesk'
+ENV['SECRET_TOKEN'] = 'd6054cf90db212c8fbc070c896c30398e3275532c5602bdf00cb153b806c000e4e46fac2f3acc0783822b8f6d30b5913b6fbcfdd24914553e745b8aa8ddfa5a4'
+ENV['DEFAULT_URL'] = 'http://www.test-url.com'
+ENV['ZENDESK_TOKEN'] = 'c30398e3275532c5602bdf00cb153b806c000e4e46fac2f3acc0783822b8f6d3'
```

Comment:
That's what I'm trying to avoid, either the test works everywhere with
exactly these settings or we have do change something.
Using ||= would also mean I can pollute my tests by accident when I
have ZENDESK_URL
etc already set.

On Fri, Mar 6, 2015 at 9:53 AM, Jon Moter <notifications@github.com> wrote:

> In config/environments/test.rb
> <https://github.com/zendesk/samson/pull/268#discussion_r25962729>:
>
> > @@ -47,3 +47,8 @@
> >
> >    config.active_support.test_order = :random
> >  end
> > +
> > +ENV['ZENDESK_URL'] = 'https://test.support.zendesk'
> > +ENV['SECRET_TOKEN'] = 'd6054cf90db212c8fbc070c896c30398e3275532c5602bdf00cb153b806c000e4e46fac2f3acc0783822b8f6d30b5913b6fbcfdd24914553e745b8aa8ddfa5a4'
> > +ENV['DEFAULT_URL'] = 'http://www.test-url.com'
> > +ENV['ZENDESK_TOKEN'] = 'c30398e3275532c5602bdf00cb153b806c000e4e46fac2f3acc0783822b8f6d3'
>
> How about using ||= for these values in case you want to override them
> when running a test?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zendesk/samson/pull/268/files#r25962729>.
>